test/files/pos/t4225.scala

```diff
@@ -0,0 +1,22 @@
+object Test {
+  class Foo {
+    class Bar
+    object Bar {
+      implicit def mkBar: Bar = new Bar
+    }
+  }
+
+  implicit class Ops[F <: Foo](val f0: F) {
```

Comment:
I've added some test variants which cover by name arguments (for the ops class and for the ops methods) and (by name) right associative ops and they don't cause any problems. Do you have a particular scenario in mind which you think might be problematic?